tensorpipe/test/test_environment.cc

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <tensorpipe/test/test_environment.h>
+
+#include <tensorpipe/config.h>
+
+#if TENSORPIPE_SUPPORTS_CUDA
+#include <cuda_runtime.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <tensorpipe/common/cuda.h>
+#include <tensorpipe/common/defs.h>
+#include <unistd.h>
+#endif // TENSORPIPE_SUPPORTS_CUDA
+
+int TestEnvironment::numCudaDevices() {
+  static int count = -1;
+  if (count == -1) {
```

Comment:
Do we need to worry about race conditions when multiple threads call this function at the same time? (The C++ runtime guarantees atomicity on the assignment of a local static variable but not on subsequent modifications).

Reply:
The race condition here is pretty harmless, as `cudaGetDeviceCount()` will always return the same value, so worst case it will be called multiple times.